staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -199,6 +202,18 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			return
 		case <-timeoutCh:
 			return
+		case <-t.C:
+			t.Reset(30 * time.Second)
+
+			outEvent := &metav1.WatchEvent{Type: string(watch.HeartBeat)}
+			if err := e.Encode(outEvent); err != nil {
+				utilruntime.HandleError(fmt.Errorf("unable to encode watch object: %v (%#v)", err, e))
+				return
+			}
+			if len(ch) == 0 {
+				flusher.Flush()
```

Comment:
How many times bigger than a TCP keepalive is this? Don't forget to include the HTTP chunk headers.

Reply:
I will use tcpdump to check exactly.

Reply:
@lavalamp 41 bytes
```
13:51:18.287131 IP localhost.http-alt > localhost.55698: Flags [P.], seq 41:82, ack 1, win 350, options [nop,nop,TS val 2185189701 ecr 2185182201], length 41: HTTP
```